ml-algorithms/src/main/java/org/opensearch/ml/engine/analysis/HFModelTokenizerFactory.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.ml.engine.analysis;
+
+import static org.opensearch.ml.engine.utils.FileUtils.calculateFileHash;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.Map;
+
+import org.apache.lucene.analysis.Tokenizer;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.env.Environment;
+import org.opensearch.index.IndexSettings;
+import org.opensearch.index.analysis.AbstractTokenizerFactory;
+import org.opensearch.ml.engine.utils.ZipUtils;
+
+import ai.djl.huggingface.tokenizers.HuggingFaceTokenizer;
+import ai.djl.training.util.DownloadUtils;
+import ai.djl.training.util.ProgressBar;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * Factory class for creating HFModelTokenizer instances.
+ * Handles the initialization and configuration of tokenizers based on index settings.
+ */
+@Log4j2
+public class HFModelTokenizerFactory extends AbstractTokenizerFactory {
+    public static final String DEFAULT_TOKENIZER_NAME = "bert-uncased";
+
+    /**
+     * Atomically loads the HF tokenizer in a lazy fashion once the outer class accesses the static final set the first time.;
+     */
+    private static abstract class BaseTokenizerHolder {
+        private static final String ZIP_PREFIX = ".zip";
+        private static final String TOKENIZER_FILE_NAME = "tokenizer.json";
+        private static final String TOKEN_WEIGHTS_FILE_NAME = "idf.json";
+
+        HuggingFaceTokenizer tokenizer;
+        Map<String, Float> token_weights;
+        String name;
+
+        BaseTokenizerHolder(String name, String url, String hashCode) {
+            this.name = name;
+
+            try {
+                Files.createDirectories(DJLUtils.getMlEngine().getAnalysisRootPath());
+                Path filePath = DJLUtils.getMlEngine().getAnalysisRootPath().resolve(name + ZIP_PREFIX);
+                downloadTokenizerFile(url, filePath, hashCode);
```

Comment:
question here, the file path is coded here, so it will download the default tokenizer for bert-uncased, can user change to use other tokenizer instead of this default tokenizer? it looks like it's not allowed at the moment.